src/entt/meta/meta.hpp

```diff
@@ -1305,6 +1347,80 @@ class meta_type {
         return construct(arguments.data(), sizeof...(Args));
     }
 
+    /**
+     * @brief Invokes the function with the given identifier, if possible.
+     *
+     * To invoke a meta function, the parameters must be such that a cast or
+     * conversion to the required types is possible. Otherwise, an empty and
+     * thus invalid wrapper is returned.<br/>
+     * It must be possible to cast the instance to the parent type of the meta
+     * function. Otherwise, invoking the underlying function results in an
+     * undefined behavior.
+     *
+     * @param id identifier of function to invoke.
+     * @param instance An opaque instance of the underlying type.
+     * @param args Parameters to use to invoke the function.
+     * @param sz Number of parameters to use to invoke the function.
+     * @return A meta any containing the returned value, if any.
+     */
+    [[nodiscard]] meta_any invoke(const id_type id, meta_handle instance, meta_any * const args, const std::size_t sz) const {
+        auto const f = func(id);
```

Comment:
I don't think it's a good idea. I see shortcuts as an instrument for writing less code. If the user has to check the validity of meta-function before invoking them through a shortcut then the shortcut doesn't function at its full power. A better solution would be using an Either type for the return value but there is no such thing in standard C++17.

Reply:
Of course, making it fail is trivial.

Reply:
> f? Really? Was there anything less explanatory than this?
`_` is a nice name) I just didn't think that at such a small scope a proper name is required. To be more precise, I'd give it a name like `maybe_func` but I thought that you'll consider it being overkill.

Reply:
> A better solution would be using an Either type for the return value but there is no such thing in standard C++17.

Maybe you are thinking of [std::optional](https://en.cppreference.com/w/cpp/utility/optional)? (Ignore me if I am wrong.)

Reply:
`std::optional` represents a value is either there or not. AFAIK, Either types represent either a valid value or an error. A closer match is `std::expected` or `boost::outcome`.

Reply:
Yeah, make sense. The _invalid_ meta any object already tells you clearly that the functions wasn't found. In all other cases, you receive a valid instance of meta any that wraps the return type, even in case of `void`.